app/src/main/java/com/example/myapplication/FirstFragment.java

```
@@ -115,16 +124,33 @@ public void afterTextChanged(Editable s) {
                 if (input != null) {
                     String matchNumber = String.valueOf(input);
                     UserModel.getMatchData().setMatchNumber(matchNumber);
+                    try {
+                        binding.input.setText(MainActivity.teams.get(Integer.parseInt(matchNumber) - 1));
+                    } catch (IndexOutOfBoundsException e){
+                        if (matchNumber.equals("0")) {
+                            binding.matchInput.setText("1");
+                            UserModel.getMatchData().setMatchNumber("1");
+                            binding.input.setText(MainActivity.teams.get(0));
+                        } else {
+                            binding.matchInput.setText(String.valueOf(MainActivity.teams.size()));
+                            UserModel.getMatchData().setMatchNumber(String.valueOf(MainActivity.teams.size()));
+                            binding.input.setText(MainActivity.teams.get(MainActivity.teams.size() - 1));
+                        }
+                    } catch (Exception e){
+
```

Again, is there something better to do than to just ignore the exception?